File: examples/python/hello-ksm.py
from keeper_secrets_manager_core import SecretsManager
from keeper_secrets_manager_core.storage import FileKeyValueStorage

if __name__ == '__main__':

    secrets_manager = SecretsManager(
        hostname='keepersecurity.com',
        token='<ONE TIME TOKEN>',
        config=FileKeyValueStorage(),
    )

    all_records = secrets_manager.get_secrets()

    for r in all_records:

        print("\tPassword: %s" % r.password)

        print("\tRecord details:\n%s" % r.dict)

        count = 0
        for f in r.files:
            count = count + 1
            print("\t\t%s file: %s" % (count, f))

            file_path = "/tmp/" + f.name

            f.save_file(file_path, create_folders=True)

File: examples/go/hello-ksm.go
package main

import (
	"fmt"

	ksm "github.com/keeper-security/secrets-manager-go/core"
)

func main() {
	// One time tokens can be used only once - afterwards use the generated config.json
	hostname := "keepersecurity.com"
	token := "<ONE TIME TOKEN>"
	sm := ksm.NewSecretsManagerFromSettings(token, hostname, true)
	// sm := ksm.NewSecretsManagerFromConfig(ksm.NewFileKeyValueStorage("client-config.json"))

	if allRecords, err := sm.GetSecrets([]string{}); err != nil {
		println("Error retrieving all records: ", err.Error())
	} else {
		for _, r := range allRecords {
			println("\tPassword: ", r.Password())
			println("\tRecord details: ", r.RawJson)

			for i, f := range r.Files {
				fmt.Printf("\t\tfile #%d -> name: %s", i, f.Name)
				filePath := "/tmp/" + f.Name
				f.SaveFile(filePath, true)
			}
		}
	}
}
